content/en/docs/setup/upgrade/istioctl-upgrade/index.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Upgrade Istio using istioctl [Experimental]
+description: Upgrade or downgrade Istio using the istioctl upgrade command.
+weight: 25
+aliases:
+    - /docs/setup/kubernetes/upgrade/steps/
+    - /docs/setup/upgrade/steps
+keywords: [kubernetes,upgrading]
+---
+
+{{< boilerplate experimental-feature-warning >}}
+
+The `istioctl experimental upgrade` command performs an upgrade of Istio. Before performing
+the upgrade, it checks that the Istio installation meets the upgrade eligibility
+criteria. Also, it alerts the user if it detects any changes in the profile
+default values between Istio versions.
+
+The upgrade command can also perform a downgrade of Istio.
+
+See the [`istioctl` upgrade reference](/docs/reference/commands/istioctl/#istioctl-experimental-upgrade)
+for all the options provided by the `istioctl experimental upgrade` command.
+
+## Upgrade prerequisites
+
+Ensure you meet these requirements before starting the upgrade process:
+
+* Istio version 1.3.3 or higher is installed.
+
+* Your Istio installation was installed using the
+   [operator CLI command](/docs/setup/install/operator/).
+
+## Upgrade steps
+
+{{< warning >}}
+Traffic disruption may occur during the upgrade process. To minimize the disruption, ensure
+that at least two replicas of each component (except Citadel) are running. Also, ensure that
+[`PodDistruptionBudgets`](https://kubernetes.io/docs/tasks/run-application/configure-pdb/)
+are configured with a minimum availability of 1.
+{{< /warning >}}
+
+The commands in this section should be run using the new version of istioctl which
+can be found in the `bin/` subdirectory of the downloaded package.
+
+1. [Download the new Istio release](/docs/setup/#downloading-the-release)
+   and change directory to the new release directory.
+
+1. Verify that `istoctl` supports upgrading from your current Istio version by
+   viewing the supported versions list:
+
+    {{< text bash >}}
+    $ istioctl manifest versions
+    {{< /text >}}
+
+1. Ensure that your Kubernetes configuration points to the cluster to upgrade:
+
+    {{< text bash >}}
+    $ kubectl config view
+    {{< /text >}}
+
+1. Begin the upgrade by running this command:
+
+    {{< text bash >}}
+    $ istioctl experimental upgrade -f `<your-custom-configuration-file>`
+    {{< /text >}}
+
+    `<your-custom-configuration-file>` is the file you used to install the
+    currently-running version of Istio.
```

Comment:
This sentence isn't clear enough. Where does one get this file from? Does it need to be generated https://deploy-preview-5245--preliminary-istio.netlify.com/docs/setup/install/operator/#inspect-modify-a-manifest-before-installation? This needs to be explained.

Reply:
This page explains installation with istioctl: 
https://preliminary.istio.io/docs/setup/install/operator/
Here's an example of using a custom configuration file: 
https://preliminary.istio.io/docs/setup/install/operator/#configure-the-feature-or-component-settings
Basically, this is like passing a yaml config file to helm template or tiller.

Reply:
I see, it's the IstioControlPlane API CRD. The doc should say that: `... is the file containing the [IstioCotrolPlane](link) you used to customize the installation of the currently-running ...`.

What if I installed without a customization resource, i.e., `istioctl manifest apply --set profile=demo`? Is the `-f` argument optional? If so, the doc should also say it's only needed if one was used during the install.

What if the install used other `--set` arguments? Don't we need to pass those into the upgrade command too?

Reply:
Yes, if you used --set (or just installed the default profile with `istioctl manifest apply` for default profile) you don't need -f, but you do need to pass the same flags as for the original install. 
I guess we hope that users don't use --set for anything that they plan to upgrade in the future - this is something we should probably emphasize somewhere.

Reply:
Yes, we can say that in the best practices documentation, but the fact that you need to use the same args here as you did during the install, also needs to be mentioned here.
^^^ @adammil2000 

Another problem is that the `istioctl x upgrade` command doc does NOT say that it supports it `--set` option. Is that a doc issue, or does it currently not support it? 
https://preliminary.istio.io/docs/reference/commands/istioctl/#istioctl-experimental-upgrade

Reply:
Upgrade does not support --set atm; there is a gap here. We could add --set or require that users install using -f if they want to use the upgrade command, or at the very least refactor their --set flags into a yaml file.
I dislike the first option because using --set just doesn't seem like very good practice for upgrades.